salalib/pointdata.cpp

```diff
@@ -3579,9 +3589,9 @@ bool PointMap::mergePoints(const Point2f& p)
    //
    PixelRef offset = pixelate(p) - PixelRef(tr.x,bl.y);
    //
-   for (size_t i = 0; i < m_selection_set.size(); i++) {
-      PixelRef a = m_selection_set[i];
-      PixelRef b = ((PixelRef)m_selection_set[i]) + offset;
+   for (auto& sel: m_selection_set) {
+      PixelRef a = sel;
+      PixelRef b = ((PixelRef)sel) + offset;

```

Comment:
Uuh, that code looks dodgy to begin with, how is PixelRef::operator + defined?

Reply:
It just adds the two:

```
inline PixelRef operator + (const PixelRef a, const PixelRef b)
{
   return PixelRef(a.x + b.x, a.y + b.y);
}
```